src/EmbedIO/Files/FileSystemProvider.cs

```diff
@@ -18,6 +19,9 @@ public class FileSystemProvider : IDisposable, IFileProvider
         /// <summary>
         /// Initializes a new instance of the <see cref="FileSystemProvider"/> class.
         /// </summary>
+        /// <remarks>
+        /// OSX doesn't support <see cref="FileSystemWatcher" />, the parameter <paramref name="isImmutable" /> will be always <see langword="false"/>.
```

Comment:
On a system that does not support `FileSystemWatcher`, `IsImmutable` must be `true`, not `false`, because we cannot detect changes.